eng/scripts/automation/GenerateAndBuildLib.ps1

```diff
@@ -357,6 +357,100 @@ function New-MgmtPackageFolder() {
+    )
+    $serviceFolder = (Join-Path $sdkPath "sdk" $service)
+    $projectFolder=(Join-Path $sdkPath "sdk" $service $namespace)
+    $ciymlFilePath =(Join-Path $sdkPath "sdk" $service $CI_YAML_FILE)
+    $apifolder = (Join-Path $projectFolder "api")
+    Write-Host "projectFolder:$projectFolder, apifolder:$apifolder"
+    if ((Test-Path -Path $projectFolder) -And (Test-Path -Path $apifolder)) {
+        Write-Host "Path exists!"
+        if (Test-Path -Path $projectFolder/src/autorest.md) {
+            Remove-Item -Path $projectFolder/src/autorest.md
+        }
+        $projFile = (Join-Path $projectFolder "src" "$namespace.csproj")
+        $fileContent = Get-Content -Path $projFile
+        $match = ($fileContent | Select-String -Pattern "<AutoRestInput>").LineNumber
+        if ($match.count -gt 0) {
+            $fileContent[$match[0] - 1] = "<AutoRestInput>$cadlInput</AutoRestInput>";
+        } else {
+            $startNum = ($fileContent | Select-String -Pattern '</PropertyGroup>').LineNumber[0]
+            $fileContent[$startNum - 2] += ([Environment]::NewLine + "<AutoRestInput>$cadlInput</AutoRestInput>")
+        }
+        $fileContent | Out-File $projFile
+
+        Update-CIYmlFile -ciFilePath $ciymlFilePath -artifact $namespace
+    } else {
+        Write-Host "Path doesn't exist. create template."
+        dotnet new -i $sdkPath/sdk/template
+        Write-Host "Create project folder $projectFolder"
+        if (Test-Path -Path $projectFolder) {
+            Remove-Item -Path $projectFolder -ItemType Directory
+        }
+
+        Push-Location $serviceFolder
+        $namespaceArray = $namespace.Split(".")
+        if ( $namespaceArray.Count -lt 3) {
+            Throw "Error: invalid namespace name."
+        }
+
+        $endIndex = $namespaceArray.Count - 2
+        $clientName = $namespaceArray[-1]
+        $groupName = $namespaceArray[1..$endIndex] -join "."
+        $dotnetNewCmd = "dotnet new azsdkdpg --name $namespace --clientName $clientName --groupName $groupName --serviceDirectory $service --force"
+        
+        if (Test-Path -Path $ciymlFilePath) {
+            Write-Host "ci.yml already exists. update it to include the new serviceDirectory."
+            Update-CIYmlFile -ciFilePath $ciymlFilePath -artifact $namespace
+
+            $dotnetNewCmd = $dotnetNewCmd + " --includeCI false"
+        }
+        # dotnet new azsdkdpg --name $namespace --clientName $clientName --groupName $groupName --serviceDirectory $service --swagger $inputfile --securityScopes $securityScope --securityHeaderName $securityHeaderName --includeCI true --force
+        Write-Host "Invoke dotnet new command: $dotnetNewCmd"
+        Invoke-Expression $dotnetNewCmd
+
+        $projFile = (Join-Path $projectFolder "src" "$namespace.csproj")
+        $fileContent = Get-Content -Path $projFile
+        $fileContent = $fileContent -replace "<Version>*.*.*-*.*</Version>", "<Version>1.0.0-beta.1</Version>"
```

Comment:
This might work, but is not safe. This is not a proper regex, which is what is expected here. Instead, probably this would be safer (and generally use single quotes in a literal string, especially when regex is involved):
```suggestion
        $fileContent = $fileContent -replace '<Version>[^<]+</Version>', '<Version>1.0.0-beta.1</Version>'
```
This would work even if the XML content were compressed (all on one line) where yours may not since it's greedy: `.*` in regex will match *everything* until the very last `</Version>` which, if there's more than one in the file, would replace *far* more content than you want.

Reply:
Thanks @heaths . I updated.